### PR TITLE
Add testset to bitarray

### LINKED ISSUE #18806: Add test sets to bit array tests
This is a huge file but pretty well commented. It's a good candidate for conversion to `@testset`. Ping me if you need help!

### PR DESCRIPTION
**Pull request by kumarkrishna**
Fixes #18806

### CODE REVIEW

**Comment by kshyatt** [comment]
cc @carlobaldassi

File: test/bitarray.jl [hunk 1]
```diff
@@ -92,109 +91,111 @@ end
 
 timesofar("conversions")
 
-## utility functions ##
+@testset "utility functions" begin
+    let b1 = bitrand(v1)
```
Comment (tkelman) [inline test/bitarray.jl]: don't need the `let` any more if it's in a testset

Comment (vtjnash) [inline test/bitarray.jl]: You need it until https://github.com/JuliaLang/julia/pull/19324 merges

Comment (tkelman) [inline test/bitarray.jl]: there's no global `b1` in this file any more

File: test/bitarray.jl [hunk 2]
```diff
@@ -735,315 +709,310 @@ let b1 = bitrand(v1)
...
+            b1 = bitrand(n1, n1)
+        end
+        b2 = bitrand(n1, n1)
+
+        @check_bit_operation (*)(b1, b2) Matrix{Int}
+        @check_bit_operation (/)(b1, b1) Matrix{Float64}
+        @check_bit_operation (\)(b1, b1) Matrix{Float64}
+
+        b0 = falses(0)
+         b0 = falses(0)
```
Comment (tkelman) [inline test/bitarray.jl]: duplicated line

File: test/bitarray.jl [hunk 3]
```diff
@@ -1266,210 +1227,207 @@ end
...
 
-## Permutedims ##
+@testset "Permutedims" begin
 
-let b1 = bitrand(s1, s2, s3, s4)
+    b1 = bitrand(s1, s2, s3, s4)
     p = randperm(4)
     @check_bit_operation permutedims(b1, p) BitArray{4}
     @check_bit_operation permutedims(b1, tuple(p...)) BitArray{4}
+
```
Comment (tkelman) [inline test/bitarray.jl]: I'd avoid blank lines right after `begin` or right before `end`

File: test/bitarray.jl [hunk 4]
```diff
@@ -1229,14 +1229,14 @@ timesofar("reductions")
 
 @testset "map over bitarrays" begin
     for l = [0, 1, 63, 64, 65, 127, 128, 129, 255, 256, 257, 6399, 6400, 6401]
-        b1 = bitrand(l)
+         b1 = bitrand(l)
```
Comment (tkelman) [inline test/bitarray.jl]: was fine before

File: test/bitarray.jl [hunk 5]
```diff
@@ -745,292 +718,281 @@ end
...
-
-# Matrix{Bool}/Matrix{Float64}
-
-let b1 = bitrand(n1, n2)
-    f2 = 1.0 .+ rand(n1, n2)
-    @check_bit_operation broadcast(*, b1, f2) Matrix{Float64}
-    @check_bit_operation broadcast(/, b1, f2) Matrix{Float64}
-    @check_bit_operation broadcast(^, b1, f2) Matrix{Float64}
-    @check_bit_operation broadcast(div, b1, f2)  Matrix{Float64}
-    @check_bit_operation broadcast(mod, b1, f2)  Matrix{Float64}
```
Comment (tkelman) [inline test/bitarray.jl]: your testset for `Matrix{Bool}/Matrix{Float64}` is missing this `mod` line

File: test/bitarray.jl [hunk 6]
```diff
@@ -1264,210 +1215,202 @@ end
...
-    @test_throws ArgumentError findmin(b1)
-end
-
-for b1 in [falses(v1), trues(v1),
-           BitArray([1,0,1,1,0]),
-           BitArray([0,0,1,1,0]),
-           bitrand(v1)]
-    @check_bit_operation findmin(b1)
-    @check_bit_operation findmax(b1)
+    @testset "findmax, findmin" begin
```
Comment (tkelman) [inline test/bitarray.jl]: findmax and findmin aren't really a subset of "Linear algebra"

**Comment by tkelman** [comment]
Is this still "WIP" ?

**Comment by kumarkrishna** [comment]
@tkelman The PR is mergeable, and the tests pass on local as well on as CI. The errors on Travis seem to be unrelated to this PR.

**Pull request merged**